src/main/lombok/no/hvl/dat250/h2020/group5/entities/Vote.java

```diff
@@ -35,37 +39,37 @@ public Vote answer(AnswerType answer) {
     return this;
   }
 
-  /**
-   * Check if poll already have the vote to avoid circular dependency.
-   *
-   * @param poll
-   * @return true if vote is added to this poll
-   */
-  public boolean setPoll(Poll poll) {
-    if (poll.getVotes().contains(this)) {
-      return false;
+  /** @param poll if poll is to be deleted it will set poll to null */
+  public void setPollAndAddThisVoteToPoll(Poll poll) {
+    if (poll == null) {
+      this.poll.getVotes().remove(this);
+      this.poll = null;
```

Comment:
Aah nå ser jeg hva som skjer. Men hadde vært lettere å forstå hvis man lagde en egen deletemetode her kanskje? :D

Reply:
Ja, det hadde muligens vært ryddigere 😄. Men samtidig skal denne metoden sette begge sidene av forholdet, så da gir det litt mening at den fjerner hele forholdet når den får null. Uansett burde jeg oppdatert javadocs 😬

Reply:
Ja, det er sant. 👍 @oddhus  Vet ikke om du har sett det, men testen `GuestControllerIT.shouldDeletePoll()` failer når man kjører i Intellij, men passer når man kjører maven. Trodde jeg fikset det i #8806beb , virker det hos deg?

Reply:
Ser at den mangler teardownen i dev

` @AfterEach
  public void tearDown() {
    for (Vote vote : voteRepository.findAll()) {
      vote.setPollOnlyOnVoteSide(null);
      vote.setVoterOnlyOnVoteSide(null);
      voteRepository.delete(vote);
    }

    guestRepository.deleteAll();
  }`